src/@context/Asset.tsx

```diff
@@ -48,136 +48,130 @@ function AssetProvider({
+  // Helper: Get and set asset access details
+  // -----------------------------------
+  const fetchAccessDetails = useCallback(async (): Promise<void> => {
+    if (!asset?.chainId || !asset?.services) return
+
+    const accessDetails = await getAccessDetails(
+      asset.chainId,
+      asset.services[0].datatokenAddress,
+      asset.services[0].timeout,
+      accountId
+    )
+    setAsset((prevState) => ({
+      ...prevState,
+      accessDetails
+    }))
+    LoggerInstance.log(`[asset] Got access details for ${did}`, accessDetails)
+  }, [asset?.chainId, asset?.services, accountId, did])
 
   // -----------------------------------
-  // Get and set assetExtended based on passed DID
+  // 1. Get and set asset based on passed DID
   // -----------------------------------
   useEffect(() => {
-    if (!did || !appConfig.metadataCacheUri) return
-
-    let isMounted = true
-
-    async function init() {
-      const assetExtended = await fetchAsset(newCancelToken())
-      if (!isMounted || !assetExtended) return
-      LoggerInstance.debug('[asset] Got assetExtended', assetExtended)
-      setAssetExtended(assetExtended)
-      setTitle(assetExtended.metadata.name)
-      setOwner(assetExtended.nft.owner)
-      setIsInPurgatory(
-        (assetExtended.purgatory?.state as unknown as string) === 'true'
-      )
-      setPurgatoryData(assetExtended.purgatory)
-    }
-    init()
-    return () => {
-      isMounted = false
-    }
-  }, [did, appConfig.metadataCacheUri, fetchAsset, newCancelToken])
+    if (!isMounted || !appConfig?.metadataCacheUri) return
+
+    fetchAsset(newCancelToken())
+  }, [appConfig?.metadataCacheUri, fetchAsset, newCancelToken, isMounted])
 
   // -----------------------------------
-  // Attach price to asset
+  // 2. Attach access details to asset
   // -----------------------------------
-  const initPrice = useCallback(
-    async (assetExtended: AssetExtended, accountId: string): Promise<void> => {
-      if (!assetExtended) return
-      const accessDetails = await getAccessDetails(
-        assetExtended.chainId,
-        assetExtended.services[0].datatokenAddress,
-        assetExtended.services[0].timeout,
-        accountId
-      )
-      assetExtended.accessDetails = accessDetails
-      setConsumeDetails({ ...accessDetails })
-      setAssetExtended(assetExtended)
-    },
-    []
-  )
-
   useEffect(() => {
-    if (!assetExtended) return
-    initPrice(assetExtended, accountId)
-  }, [accountId, assetExtended, initPrice])
+    if (!isMounted) return
+
+    fetchAccessDetails()
+  }, [accountId, fetchAccessDetails, isMounted])
 
   // -----------------------------------
   // Check user network against asset network
   // -----------------------------------
   useEffect(() => {
-    if (!networkId || !assetExtended) return
+    if (!networkId || !asset) return
 
-    const isAssetNetwork = networkId === assetExtended?.chainId
+    const isAssetNetwork = networkId === asset?.chainId
```

Comment:
should we stop using both `networkId` and `chainId` and stick with one? In theory there are differences, but we use it with the same meaning . Not sure if relevant to this exact pr

Reply:
it kinda depends on what exactly Aquarius means with `chainId`. 

Just seeing that in our web3 context provider we catch both, `chainId` (`web3.eth.getChainId()`) & `networkId` (`web3.eth.net.getId()`)  so we in market codebase we can simply switch to use `chainId` instead. But we need to double check if it's the same for Aquarius.

Reply:
switching to `chainId` in this context worked so just did it in here. Won't touch any other place for now